test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/io/CharArrayReaderTest.scala

```diff
@@ -0,0 +1,182 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.javalib.io
+
+import java.io._
+import org.junit.Assert._
+import org.junit.Test
+import org.scalajs.testsuite.utils.AssertThrows._
+
+class CharArrayReaderTest {
+  private[io] val hw: Array[Char] = Array('H', 'e', 'l', 'l', 'o', 'W', 'o', 'r', 'l', 'd')
+
+  private def withClose[T <: AutoCloseable](closeable: T)(fn: T => Unit): Unit = {
+    fn(closeable)
+
+    if (closeable != null) closeable.close()
+  }
+
+  /**
+    * @tests java.io.CharArrayReader#CharArrayReader(char[])
+    */
+  @Test def should_support_constructor_char_array(): Unit = withClose(new CharArrayReader(hw)) { cr =>
+    assertTrue("Failed to create reader", cr.ready())
+  }
+
+  /**
+    * @tests java.io.CharArrayReader#CharArrayReader(char[], int, int)
+    */
+  @Test def should_support_constructor_char_array_with_offset(): Unit =
+    withClose(new CharArrayReader(hw, 5, 5)) { cr =>
+      assertTrue("Failed to create reader", cr.ready())
+      val c = cr.read()
+      assertTrue("Created incorrect reader--returned '" + c.toChar + "' intsead of 'W'", c == 'W')
```

Comment:
This seems like a use case for `assertEquals`:
```suggestion
      assertEquals('W', c)
```